Mathlib/RingTheory/Noetherian.lean

```diff
@@ -456,14 +456,167 @@ lemma LinearMap.eventually_iSup_ker_pow_eq (f : M →ₗ[R] M) :
   · rw [← hn _ (hm.trans h), hn _ hm]
   · exact f.iterateKer.monotone h.le
 
+namespace Orzech
+
+variable {R M : Type*} [Semiring R] [AddCommMonoid M] [Module R M] {N : Submodule R M}
+  (f : N →ₗ[R] M) (hf : Surjective f) (n : ℕ)
+
+open scoped Classical in
```

Comment:
This is kind of dual of `LinearMap.tunnel`. I would say that deserves a similar API.

Reply:
(This comment corrects an earlier comment, about whether one can use docs#LinearMap.tunnel. Since it is dual, probably not, but it is probably interesting to define functions similar to it for surjective linear maps. (What name to give? Funnel? lennut?)

Reply:
Could be called `cotunnel` :D

By the way, I'm not investigated whether the proof can be written using the existing `LinearMap.tunnel`.

Reply:
Done. I removed `tunnel` and `tailing`, added `iterateMapComap` for the substitution.